### PR TITLE
Make test setup functions terminate on error

### PR DESCRIPTION
**Pull request by mbknust**
I changed test setup and teardown functions to take no arguments and return void, because that's what pw_unit_test expects. If an error occurs in the initialization / deinitialization code, I call `VerifyOrDie` instead of returning an error value.

**Followup**: `VerifyOrDie(...)` should be replaced by `ASSERT_SOMETHING(...)` once transition to pw_unit_test is completed.

EDIT: Following suggestions by @andy31415, the scope was reduced - I will make the necessary methods static and rename some context classes in followup PRs.

### CODE REVIEW

**Review by arkq — APPROVED** [review]

File: src/transport/tests/LoopbackTransportManager.h [hunk 1]
```diff
@@ -27,26 +27,26 @@ class LoopbackTransportManager
 {
 public:
     /// Initialize the underlying layers.
-    CHIP_ERROR Init()
+    static CHIP_ERROR Init()
```
Comment (andy31415) [inline src/transport/tests/LoopbackTransportManager.h:30]: This class was not meant to be static. If you need static methods, please separate them somewhere else. Do not use a class as a `namespace` to place everything static inside it.

**Review by andy31415 — CHANGES_REQUESTED** [review]
LoopackTransportManager changes are suspect: we should not convert a class to an odd singleton with all methods static.

File: src/app/tests/TestEventLoggingNoUTCTime.cpp [hunk 2]
```diff
@@ -80,57 +80,56 @@ class MockClock : public chip::System::Clock::ClockBase
     chip::System::Clock::ClockBase & mRealClock;
 };
 
-class TestContext : public chip::Test::AppContext
+class TestContext : public chip::Test::AppTestFixture
 {
 public:
     // Performs shared setup for all tests in the test suite
-    CHIP_ERROR SetUpTestSuite() override
+    static void SetUpTestSuite()
```
Comment (andy31415) [inline src/app/tests/TestEventLoggingNoUTCTime.cpp:87]: make this call a static method if that is what is needed. Converting a subset of methods to be static is odd ... there is no clear reason of why some are static and some are not.

Maybe we should break down this PR into smaller chunks to define a good pattern on splitting things.

Comment (andy31415) [reply src/app/tests/TestEventLoggingNoUTCTime.cpp:87]: I see googletest style wants these static. Marking resolved.

File: src/transport/tests/LoopbackTransportManager.h (hunk 1 above)
Comment (mbknust) [reply src/transport/tests/LoopbackTransportManager.h]: Ok, I will keep this in mind when switching to static in followup PR

**Comment by cecille** [comment]
@Alami-Amine - FYI

**Review by andy31415 — APPROVED** [review]

**Pull request merged**